### PR TITLE
improve manager dashboard: add History, Home and Spend views

### PR DESCRIPTION
**Pull request by edouardparis**

### CODE REVIEW

File: src/revaultd/model.rs [hunk 1]
```diff
@@ -68,33 +68,70 @@ pub enum VaultStatus {
...
+        if let VaultTransaction::Broadcasted(tx) = &self.emergency {
+            return tx;
+        }
+
+        if let VaultTransaction::Broadcasted(tx) = &self.unvault {
+            return tx;
+        }
+
+        &self.deposit
+    }
```
Comment (darosior) [inline src/revaultd/model.rs]: There are a lot of edge cases here, but i guess it's fine

**Review by darosior — COMMENTED** [review]
Reviewed until b1ec995651c0dded024a3c920c7aeec880e7fde5, now gonna test locally and review more globally.

As ussual, feel free to ignore the nits!

**Review by darosior — APPROVED** [review]
Ok, a new bunch of nits from a new review round. I let you decide whether to apply them or not. Been playing with it and the result is good we are definitely making progress! I think i'm starting to finally get my head around iced and the overall code architecture. 

One thing still is about the documentation. It's quite hard to come review a month and a half worth of work, +2676 - 388 modifications **overall** (far less than the addition of each commit diff) without any rationale nor comment whatsoever in 33 commits. Actually it's taking *way* more time to only understand the intent of the author that there is not much left for testing.. 

I don't like useless comments either but there is a middleground to find, you could for example [briefly present the structure and intent of each module](https://github.com/re-vault/revault-gui/issues/19), add small comments for the casual reader that is not familiar with iced, and comment choices (panics, data structures (why are you using this field for?, etc..).

Regarding the commits, a clear title describing the purpose and optionally a rationale as per why making this change etc would be really welcome :) See https://github.com/re-vault/revault_tx/blob/master/CONTRIBUTING.md for more guidelines..

ACK 048d1cb -- code reviewed most of the commits and tested locally.

File: src/ui/view/manager.rs [hunk 2]
```diff
@@ -80,41 +87,98 @@ fn balance_view<'a, T: 'a>(balance: u64) -> Container<'a, T> {
...
+                                    .push(badge::tx_deposit())
+                                    .push(text::small(&self.vault.txid))
+                                    .spacing(20),
+                            )
+                            .width(Length::Fill),
+                        )
+                        .push(
+                            Container::new(Text::new(format!(
+                                "{}",
+                                self.vault.amount as f64 / 100000000_f64
```
[the remark this answers is not in the document]
Comment (edouardparis) [reply src/ui/view/manager.rs]: we display 5.22 BTC and not 522000000 satoshis

File: src/revaultd/mod.rs [hunk 3]
```diff
@@ -102,10 +102,13 @@ impl RevaultD {
...
         outpoints: Option<Vec<String>>,
     ) -> Result<ListTransactionsResponse, RevaultDError> {
-        self.call(
-            "listtransactions",
-            Some(ListTransactionsRequest { outpoints }),
-        )
+        match outpoints {
+            Some(list) => self.call(
+                "listtransactions",
+                Some(vec![ListTransactionsRequest(list)]),
```
[the remark this answers is not in the document]
Comment (edouardparis) [reply src/revaultd/mod.rs:117]: sadly does not work:
```
[2021-02-09][15:49:00][rpc][TRACE] Request: {"method":"listtransactions","params":[[["b7b345589e3eaa6d672e73ccf37b844b72475423e9fe5e62f0a7e227b0cb6cb2:1"]]],"id":135462,"jsonrpc":"2.0"}.
[2021-02-09][15:49:00][rpc][DEBUG] Response: {"jsonrpc":"2.0","error":{"code":-32602,"message":"Invalid params: invalid type: sequence, expected a string."},"id":135462}.
```

File: src/revaultd/mod.rs [hunk 4]
```diff
@@ -68,6 +68,15 @@ impl RevaultD {
...
     }
 
+    pub fn network(&self) -> bitcoin::Network {
+        match self.config.bitcoind_config.network.as_str() {
+            "bitcoin" => bitcoin::Network::Bitcoin,
+            "regtest" => bitcoin::Network::Regtest,
+            "mainnet" => bitcoin::Network::Bitcoin,
+            _ => bitcoin::Network::Bitcoin,
+        }
+    }
```
[the remark this answers is not in the document]
Comment (edouardparis) [reply src/revaultd/mod.rs:78]: we do not have the same convention as bitcoin::Network, ex: `bitcoin` and `mainnet`

File: src/ui/icon.rs [hunk 5]
```diff
@@ -0,0 +1,58 @@
+use iced::{Font, HorizontalAlignment, Length, Text};
+
+const ICONS: Font = Font::External {
+    name: "Icons",
+    bytes: include_bytes!("../../static/icons/bootstrap-icons.ttf"),
+};
```
[the remark this answers is not in the document]
Comment (edouardparis) [reply src/ui/icon.rs:6]: icons are stored as font, but I dont want to be confused with the real fonts we are using.

[the remark this answers is not in the document]
Comment (darosior) [reply src/ui/icon.rs:6]: Oh my bad i overlooked and confused it with a font

File: src/revaultd/mod.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (darosior) [reply src/revaultd/mod.rs:78]: Where do we have `mainnet` ?

[the remark this answers is not in the document]
Comment (darosior) [reply src/revaultd/mod.rs:78]: Also, this does not handle `testnet` nor upcoming `signet`. So i think this should just use rust-bitcoin's `FromStr` and fail if it is not a valid network (revaultd would never send an invalid one though)

**Comment by darosior** [comment]
ACK 098d965e69a7b8f09beda5e7cd43c9b51c5a317b
